src/data/store.rs

```diff
@@ -172,8 +160,15 @@ impl Store {
             }
 
             match entity_id {
-                ChatEntityId::Agent(agent) => {
-                    chat.send_message_to_agent(*agent, prompt, &self.mofa_backend);
+                ChatEntityId::Agent(agent_id) => {
+                    if let (Some(client), Some(agent)) = (
+                        self.chats.get_client_for_agent(agent_id),
+                        self.chats.available_agents.get(agent_id)
+                    ) {
+                        chat.send_message_to_agent(agent, prompt, &client);
+                    } else {
+                        println!("client or agent not found: {:?}", agent_id);
```

Comment:
nitpick: I would print warnings, errors and debug messages to `stderr` with `eprintln!`.